File: python3/shift_2d_grid.py
"""
Space   : O(m*n)
Time    : O(m*n)
"""


class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        row = len(grid)
        col = len(grid[0])
        temp = []

        k %= row * col

        for x in grid:
            temp.extend(x)

        temp = temp[-k:] + temp[:-k]

        p = 0
        for y in range(row):
            for x in range(col):
                grid[y][x] = temp[p]
                p += 1

        return grid

File: golang/shift_2d_grid.go
package main

// Space   : O(m*n)
// Time    : O(m*n)

func shiftGrid(grid [][]int, k int) [][]int {
    row := len(grid)
    col := len(grid[0])
    nm := row * col
    k %= nm
    var temp []int
    
    for _, x := range grid {
        temp = append(temp, x...)
    }
    temp = append(temp[nm-k:], temp[:nm-k]...)

    
    p := 0
    for y := 0; y < row; y++ {
        for x := 0; x < col; x++ {
            grid[y][x] = temp[p]
            p++
        }
    }
    
    return grid
}
